cpp/dolfinx/mesh/permutationcomputation.cpp

```diff
@@ -21,205 +23,185 @@ using namespace dolfinx;
-        if (e_vertices[v] < e_vertices[min_v])
-          min_v = v;
-
-      // pre is the number of the next vertex clockwise from the lowest
-      // numbered vertex
-      const int pre = min_v == 0 ? e_vertices[3 - 1] : e_vertices[min_v - 1];
-
-      // post is the number of the next vertex anticlockwise from the
-      // lowest numbered vertex
-      const int post = min_v == 3 - 1 ? e_vertices[0] : e_vertices[min_v + 1];
-
-      std::uint8_t g_min_v = 0;
-      for (int v = 1; v < 3; ++v)
-        if (vertices[v] < vertices[g_min_v])
-          g_min_v = v;
-
-      // pre is the number of the next vertex clockwise from the lowest
-      // numbered vertex
-      const int g_pre = g_min_v == 0 ? vertices[3 - 1] : vertices[g_min_v - 1];
-
-      // post is the number of the next vertex anticlockwise from the
-      // lowest numbered vertex
-      const int g_post = g_min_v == 3 - 1 ? vertices[0] : vertices[g_min_v + 1];
-
-      std::uint8_t rots = 0;
-      if (g_post > g_pre)
-        rots = min_v <= g_min_v ? g_min_v - min_v : g_min_v + 3 - min_v;
-      else
-        rots = g_min_v <= min_v ? min_v - g_min_v : min_v + 3 - g_min_v;
-
-      face_perm[c][3 * i] = (post > pre) == (g_post < g_pre);
-      face_perm[c][3 * i + 1] = rots % 2;
-      face_perm[c][3 * i + 2] = rots / 2;
-    }
-  }
+  std::uint8_t g_min_v = std::distance(
+      vertices.begin(), std::min_element(vertices.begin(), vertices.end()));
 
-  return face_perm;
+  // g_pre is the (global) number of the next vertex clockwise from the lowest
+  // numbered vertex
+  const int g_pre = vertices[(g_min_v + 2) % 3];
+
+  // g_post is the (global) number of the next vertex anticlockwise from the
+  // lowest numbered vertex
+  const int g_post = vertices[(g_min_v + 1) % 3];
+
+  std::uint8_t rots = 0;
+  if (g_post > g_pre)
+    rots = (g_min_v + 3 - min_v) % 3;
+  else
+    rots = (min_v + 3 - g_min_v) % 3;
+
+  return {(post > pre) == (g_post < g_pre), rots};
+}
+//-----------------------------------------------------------------------------
+std::pair<std::int8_t, std::int8_t>
+compute_quad_rot_reflect(const std::vector<std::int32_t>& e_vertices,
+                         const std::vector<std::int64_t>& vertices)
+{
+  // Find minimum local cell vertex on facet
+  std::uint8_t min_v
+      = std::distance(e_vertices.begin(),
+                      std::min_element(e_vertices.begin(), e_vertices.end()));
+
+  // Table of next and previous vertices
+  // 0 - 2
+  // |   |
+  // 1 - 3
+  const std::array<std::int8_t, 4> prev = {2, 0, 3, 1};
+
+  // pre is the (local) number of the next vertex clockwise from the
+  // lowest numbered vertex
+  std::int32_t pre = e_vertices[prev[min_v]];
+
+  // post is the (local) number of the next vertex anticlockwise
+  // from the lowest numbered vertex
+  std::int32_t post = e_vertices[prev[3 - min_v]];
+
+  // If min_v is 2 or 3, swap. Why?
+  if (min_v == 2 or min_v == 3)
+    min_v = 5 - min_v;
```

Comment:
To answer the why: min_v wants to be the number of vertices anticlockwise away the minimum numbered vertex is. If the vertices were numbered:

```
3 -- 2
|    |
0 -- 1
```

We wouldn't need to do the swap, but as we use the numbering:

```
2 -- 3
|    |
0 -- 1
```

we therefore need the swap

Reply:
Thanks @mscroggs . Maybe you can add something in-line to document this.